### PR TITLE
Ban specific tag names

### PR DESCRIPTION
**Pull request by weiquany**
#63 
Ban specific tag names to integrate well with the show command and command aliasing.

### CODE REVIEW

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 1]
```diff
@@ -11,6 +12,8 @@
  */
 public class Tag {
 
+    public static final String[] BANNED_TAG_NAMES = new String[]{"create", "c", "remove", "r"};
```
Comment (kelvinou01) [inline src/main/java/seedu/address/model/tag/Tag.java:15]: Should we define `BANNED_TAG_NAMES` in `TagCommandGroup` instead? To keep coupling low.

File: src/main/java/seedu/address/logic/parser/TagCommandParser.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public TagCommand parse(String args) throws ParseException {
             throw new ParseException(TagCommand.MESSAGE_USAGE);
         }
 
-        return index.isPresent() ? new TagCommand(index.get(), tags) : new TagCommand(tags);
+        return index.map(i -> new TagCommand(i, tags)).orElse(new TagCommand(tags));
```
Comment (kelvinou01) [inline src/main/java/seedu/address/logic/parser/TagCommandParser.java:46]: Good catch!

File: src/main/java/seedu/address/model/tag/Tag.java (hunk 1 above)
Comment (weiquany) [reply src/main/java/seedu/address/model/tag/Tag.java:15]: Good point!

**Review by malwaregarry — APPROVED** [review]
LGTM

**Review by kelvinou01 — APPROVED** [review]
LGTM!

**Pull request merged**